integration/test/app/basic/simple_test.go

```diff
@@ -212,25 +212,14 @@ func TestAppLifecycle(t *testing.T) {
 	}
 
 	{
-		config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("deleting release %#q", key.CustomResourceReleaseName()))
+		config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("deleting %#q app CR", key.TestAppReleaseName()))
 
-		err := config.Release.Delete(ctx, key.CustomResourceReleaseName())
+		err = config.Host.G8sClient().ApplicationV1alpha1().Apps(namespace).Delete(key.TestAppReleaseName(), &metav1.DeleteOptions{})
```

Comment:
This is for fixing the issue that entire configmap, secret and appCatalog deleted before we tried to delete `chart` CR.

Reply:
I think the operator needs to be able to handle this. Don't we have logic in the configmap resource etc that checks if the app CR is deleted?

Reply:
If we deleted the entire release, helm would delete related configmaps, secrets and appCatalogs. 
But if helm deletes other resources earlier than `app` CR in deletion event, app-operator could not find the resources related to this release, so it hangs. 
It happened time by time and break e2etest some cases.

Reply:
I think that's a bug then. We can't 100% control the order of deletion and if this happens in production there may be alerts.

When the app CR is deleted the operator needs to handle these cases. Please create a bug issue and link another TODO here. So we can get it added to our team board and then fixed.